cmd/all-in-one/all_in_one_test.go

```diff
@@ -166,7 +166,7 @@ func getAPITrace(t *testing.T) {
 	}
 
 	require.Len(t, queryResponse.Data, 1)
-	require.Len(t, queryResponse.Data[0].Spans, 1)
+	require.Len(t, queryResponse.Data[0].Spans, 2)
```

Comment:
@yurishkuro OTEL is now sampling two spans for `/api/services` - is this valid?

**Before**
<img width="1462" alt="Screenshot 2024-10-25 at 11 04 18 PM" src="https://github.com/user-attachments/assets/6e7b8971-4d84-43b8-babe-4f056ebb7e1e">

**After**
<img width="1462" alt="Screenshot 2024-10-25 at 11 23 37 PM" src="https://github.com/user-attachments/assets/91bb049f-e0e2-4a52-8a59-f1a762db8e9a">

Reply:
In the after screenshot above, it seems like the second span is coming from the traceMiddleware because it has the `handleFunc` because it has the `http.Route` tag on the span

Reply:
Are we dual-installing tracing instrumentation? The OTEL helper probably does it already and we don't need our own.

Reply:
@yurishkuro if you look at the diff - i removed it from `http_handler.go`. do I need to remove it from somewhere else as well?

Reply:
Don't know, but the two spans look like a bug 🐞